src/components/generator/generatorFunction.js

```diff
@@ -1,131 +1,147 @@
 import arePositionsSimilar from '../../logic/combos/similarPositions';
 
-export function findCombo(vars, retries) {
+/**
+ * Heuristically generates a random combo which adheres to the specified conditions.
+ * @param {*} conditions Conditions that the combo must meet
+ * @param {number} retries Number of times to retry when generating a combo fails
+ */
+export function generateRandomCombo(conditions, retries) {
```

Comment:
Accepting a long tuple (conditions) is a bad pattern. It would be better to accept a
config object that overrides some sensible defaults.

`generateRandomCombo` could accept an object like this
```js
const config =  {
    tricks: undefined,
    positions: undefined,
    numberOfTricks: 10,
    startFromCheckbox: false,
    // Rename! not a boolean
    // eg startingPositionIndex
    startFromPosition: 0,
    allowDuplicates: false,
    allowConsecutiveTricks: false,
    allowSimilarPositions: true,
    allowTransitions: false,
    // Rename. Does not sound like
    // a boolean
    // Also. name rather confusing
    // shouldFinishToFeet
    finishToFeet: true,
    avgDifficulty: -1,
    maxDifficulty: 11,
    ...userProvidedConfig
}

```

Reply:
Accepted